packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+// @ts-check
+
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { fit, M, makeCopyBag, makeStore } from '@agoric/store';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X } = assert;
+
+export const KW = harden({
+  /** seat keyword for use in offers to return an attestation. */
+  Attestation: 'Attestation',
+});
+
+/**
+ * Find-or-create value in store.
+ *
+ * @type {<K, V>(store: Store<K, V>, key: K, thunk: () => V) => V}
+ */
+const getOrElse = (store, key, make) => {
```

Comment:
I confess I don't understand what you are asking.

Reply:
This PR defines a `getOrElse` function that lets the caller say "give me the value for this key from the store, or if it's not there define it in the store". That seems generally useful so I asked if the Store API might offer that in the future. If not the `MapStore` type then perhaps a helper like this within `@agoric/store` ?

Reply:
I don't think there are any plans for anything like this.  The authority to consult would be @erights, who I tentatively predict, on the basis of no knowledge whatsoever on my part, will have a subtle and nuanced argument why doing this is would be a bad idea.

Reply:
@dtribble I remember you had a good verb to mean something like this `getOrElse`. Do you remember what it was?

@FUDCo I will try to avoid taking that as a challenge ;) .

I would be fine with a helper. I am reluctant to extend the type itself, but I do not currently have any principled objection. Just a general sense that the object's own API should remain reasonably small and any addition must pay for itself.

Whether helper or method, should the `make` function take the key as a parameter?

Reply:
> should the `make` function take the key as a parameter?

Interesting idea. I can see some utility. The design here is inspired by ELib, where [fetch/2](https://github.com/kpreid/e-on-java/blob/master/src/jsrc/org/erights/e/elib/tables/EMap.java#L307) took a [Thunk](https://github.com/kpreid/e-on-java/blob/master/src/jsrc/org/erights/e/elib/base/Thunk.java), which takes no args.

For reference, monte adopted the same design:
https://monte.readthedocs.io/en/latest/runtime.html#safeScope.Map.fetch
https://github.com/monte-language/typhon/blob/master/typhon/objects/collections/maps.py#L138